src/main/java/edu/ucla/library/sinai/handlers/LoginHandler.java

```diff
@@ -30,24 +31,43 @@ public LoginHandler(final Configuration aConfig, final JWTAuth aJwtAuth) {
     @Override
     public void handle(final RoutingContext aContext) {
         final MultiMap params = aContext.request().params();
+        final HttpServerResponse response = aContext.response();
 
         if (params.contains(TOKEN)) {
             final String rawToken = params.get(TOKEN);
-            final JWTOptions jwtOptions = new JWTOptions().setExpiresInSeconds(60);
-            final String token = myJwtAuth.generateToken(new JsonObject().put("sub", "asdf"), jwtOptions);
             final Cookie cookie = aContext.getCookie(TOKEN);
 
-            // Not perfect, we know, but a good faith effort
             if (cookie != null && rawToken != null && rawToken.equals(cookie.getValue())) {
-                myJwtAuth.authenticate(new JsonObject().put("jwt", token), authHandler -> {
-                    final HttpServerResponse response = aContext.response();
+                /**
+                 * At this point, we trust that the sinai-id service has authenticated the user because it has sent back
+                 * the same token that exists as a cookie in the user's browser. (That cookie is created/updated
+                 * whenever <code>src/main/webapp/templates/header.hbs</code> is rendered while the user is not logged
+                 * in, but is only used on an initial authentication request, in which it is sent to sinai-id.)
+                 *
+                 * Because the interaction with sinai-id is an ad-hoc authentication flow that doesn't correspond to any
+                 * of Vert.x's built-in AuthProviders, the following JWT authentication flow is essentially a no-op that
+                 * is necessary for one thing only: putting a User on the RoutingContext. In a sense, it implements a
+                 * custom AuthProvider for sinai-id.
```

Comment:
> In a sense, it implements a custom AuthProvider for sinai-id

re-reading this, I'm not sure it really makes sense or is even accurate / non-confusing